subsys/bluetooth/host/ecc.c

```diff
@@ -64,8 +64,18 @@ int bt_pub_key_gen(struct bt_pub_key_cb *new_cb)
 		}
 	}
 
-	new_cb->_next = pub_key_cb;
-	pub_key_cb = new_cb;
+	if (new_cb) {
+		struct bt_pub_key_cb *cb;
+
+		SYS_SLIST_FOR_EACH_CONTAINER(&pub_key_cb_slist, cb, node) {
+			if (cb == new_cb) {
+				BT_WARN("Callback already registered");
```

Comment:
This means that once cannot call bt_pub_key_gen() directly from result callback.  I'm fine with this as long as it is documented.